File: part1-coding/cpp-tests/test_merge.cpp
#include <bits/stdc++.h>
using namespace std;
vector<vector<int>> merge(vector<vector<int>>& in);
int main(){
    vector<vector<int>> in={{1,3},{2,6},{8,10},{15,18}};
    auto out=merge(in);
    assert(out.size()==3 && out[0][0]==1 && out[0][1]==6);
    cout<<"C++ merge OK\n";
    return 0;
}

File: part1-coding/04-intervals-merge/python/tests/test_merge.py
from importlib.machinery import SourceFileLoader
import os
mod = SourceFileLoader('merge', os.path.join(os.path.dirname(__file__), '..', '04_intervals_merge.py')).load_module()

def test_merge_intervals():
    iv = [[1,3],[2,6],[8,10],[15,18]]
    assert mod.merge(iv) == [[1,6],[8,10],[15,18]]
